docs/config/ssr-options.md

```diff
@@ -2,17 +2,17 @@
 
 ## ssr.external
 
-- **Type:** `string[]`
+- **Type:** `string[] | true`
 - **Related:** [SSR Externals](/guide/ssr#ssr-externals)
 
-Force externalize dependencies for SSR.
+Externalize dependencies for SSR. By default, all dependencies are externalized, except linked dependencies for HMR. This can opted-out by adding the dependency to this option, or set `true` to force externalize all dependencies.
 
 ## ssr.noExternal
 
 - **Type:** `string | RegExp | (string | RegExp)[] | true`
 - **Related:** [SSR Externals](/guide/ssr#ssr-externals)
 
-Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized.
+Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized. This option has a higher priority than `ssr.external` except if a dependency is explicitly specified in the `ssr.external` array.
```

Comment:
The new sentence is not clear to me

Let's say a depends on b depends on c.

If a and b are in noExternal and b is in external, which are externalized?
If a is in noExternal and b is in external, which are externalized?

Reply:
> If a and b are in noExternal and b is in external, which are externalized?

If `a` and `b` are explicitly configured in `external` and `noExternal` configs, `a` will be noExternalized, and `b` will be externalized.

If `a` and `b` are explicitly configured in `noExternal` configs, `external` config is unconfigured because `b` is already external by default, `a` and `b` will be noExternalized. 



> If a is in noExternal and b is in external, which are externalized?

`a` will be noExternalized, `b` will be externalized.

---

The priority order it something like, (highest to lowest priority) explicit external > explicit noExternal > (is linked dep ? default noExternal : default external).

Reply:
```suggestion
Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized. If a dependency is explicitly defined in `ssr.external`, it will take priority and be externalized instead.
```

maybe something like this is clearer?

Reply:
just to make sure I understand, can you clarify what will happen to `c` in my two examples?

Reply:
in both examples because `b` is externalized, `c` will also be externalized as Vite can't touch `b`'s code.



> If `a` and `b` are explicitly configured in `noExternal` configs, `external` config is unconfigured because `b` is already external by default, `a` and `b` will be noExternalized.

For this case where `b` is `noExternalized`, and `c` isn't explicitly configured anywhere, `c` will also be noExternalized otherwise it would error in runtime in strict dependency installations, because your app in root can't import `c`.

Reply:
yeah, I like your new suggested version of the docs. just one minor tweak I'd make to it:
```suggestion
Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized. If a dependency is explicitly defined in `ssr.external`, it will take priority and be externalized along with its transitive dependencies.
```